### ISSUE TITLE
No warning for a class implementing itself

### ISSUE THREAD

**Issue by lauraharker** [microsoft/TypeScript#45806]
# Bug Report

### 🔎 Search Terms

[implements itself, self implements]

### 🕗 Version & Regression Information

- This is the behavior in every version I tried from 3.3 on, and I reviewed the FAQ for entries about inheritance.

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.4.2#code/MYGwhgzhAEBiD29oEsC2AHEBTVWB2ALjAkgN4C+QA)

### 💻 Code

```ts
class Foo implements Foo {}
```

### 🙁 Actual behavior

This code passed typechecking.

### 🙂 Expected behavior

I expected the TypeScript compiler to emit an error on this code.

As far as I know, it's a no-op for a class to implement itself. The code author very likely intended to write something else.

This came up in the wild in TypeScript inside Google's repo. A file was trying to create a fake version of [`MediaSession`](https://developer.mozilla.org/en-US/docs/Web/API/MediaSession) with `export class MediaSession implements MediaSession {` for testing. The class accidentally implemented itself instead of the `MediaSession` interface from https://github.com/DefinitelyTyped/DefinitelyTyped/blob/c6f1fc8c0b7c2ebe8e099586af618b0a33cfef65/types/wicg-mediasession/index.d.ts#L20.

Another example found via Google: https://stackoverflow.com/questions/54306190/what-does-it-mean-when-a-class-implements-itself-in-typescript
